src/cpp/session/modules/SessionFind.cpp

```diff
@@ -1753,7 +1757,11 @@ core::Error Replacer::replaceRegexIgnoreCase(size_t matchOn, size_t matchOff,
 {
    try
    {
-      boost::regex find(findRegex, boost::regex::grep | boost::regex::icase);
+
+      // egrep (grep -E) flavor regex includes POSIX ERE
+      boost::regex_constants::syntax_option_type flags = boost::regex::egrep & boost::regex::bk_vbar;
+      flags = flags | boost::regex::icase;
```

Comment:
I wasn't 100% sure about the difference between `&` and `|` in terms of these flags, how they're combined, and their precedence, but this seemed in line with other usages I saw for these flags

Reply:
These are bitflags; those are usually combined with `|`. `&` is usually used to check if a particular bitflag is "active" in a set.

Reply:
Hm, using `&` here means we're not actually passing either flag to our regex and are relying on defaults (perl regexes). You'd get the same result if you only set `boost::regex::icase`.

Reply:
That is surprising to me. You can see how flags are defined and combined here, for example, in the boost.regex source code:

https://github.com/boostorg/regex/blob/a9f18c2c03a5e64745ec8fcdb8247dbb2a84e9df/include/boost/regex/v4/regbase.hpp#L93-L104

I wonder if we want to start with "normal" and then add some flags to turn off PCRE-specific features? (Or, just use `perl` anyway?)

Reply:
Relying on PCRE seems reasonable to me, especially since either way we were announcing a change in behavior and all of our unit tests pass on all platforms. Is that what R uses?

Reply:
I think not by default -- R's grep functions all take a `perl` argument, but it's FALSE by default